src/c4/ofpstream.cc

```diff
@@ -52,8 +52,8 @@ void ofpstream::mpibuf::send() {
     for (unsigned i = 1; i < pids; ++i) {
       unsigned N(0);
       receive(&N, 1, i);
+      buffer_.resize(N);
```

Comment:
Move the resize call so that it is always made, even when the rank has no output (N==0).